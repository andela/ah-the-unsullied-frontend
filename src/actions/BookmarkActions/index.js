import { toast } from 'react-toastify';
import {
  BOOKMARK_ARTICLE,
  GET_BOOKMARK_ARTICLES,
  GET_BOOKMARK_ERRORS
} from '../actionTypes';
import axiosConfig from '../../axiosConfig';

export const bookmarkArticle = slug => dispatch => {
  return axiosConfig
    .request({
      method: 'post',
      url: 'articles/' + slug + '/bookmark'
    })
    .then(res => {
      toast.success(res.data.message);
      dispatch({
        type: BOOKMARK_ARTICLE,
        payload: res.data.message
      });
    })
    .catch(err => {
      toast.error(err.response.data['detail']);
      dispatch({
        type: GET_BOOKMARK_ERRORS,
        payload: err.response.data
      });
    });
};

export const getBookmarkArticles = () => dispatch => {
  return axiosConfig
    .request({
      method: 'get',
      url: 'articles/bookmark/'
    })
    .then(res => {
      dispatch({
        type: GET_BOOKMARK_ARTICLES,
        payload: res.data
      });
    })
    .catch(err => {
      dispatch({
        type: GET_BOOKMARK_ERRORS,
        payload: err.response.data
      });
    });
};
