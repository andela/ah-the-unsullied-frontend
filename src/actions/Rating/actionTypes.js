export const POST_RATING = 'POST_RATING';
export const VIEW_AVERAGE_RATING = 'GET_AVERAGE_RATING';
export const RATING_ERRORS = 'RATING_ERRORS';
