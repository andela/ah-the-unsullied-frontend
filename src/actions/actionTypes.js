export const SET_CURRENT_USER = 'SET_CURRENT_USER';
export const GET_ERRORS = 'GET_ERRORS';
export const SIGNUP_USER = 'SIGNUP_USER';
export const SIGNUP_ERROR = 'SIGNUP_ERROR';
export const VERIFY_USER = 'VERIFY_USER';
export const VERIFY_ERROR = 'VERIFY_ERROR';
export const UPDATE_COUNT = 'UPDATE_COUNT';
export const PASSWORD_RESET = 'PASSWORD_RESET';
export const PASSWORD_RESET_ERROR = 'PASSWORD_RESET_ERROR';
export const SEND_EMAIL_ERROR = 'SEND_EMAIL_ERROR';
export const SEND_EMAIL = 'SEND_EMAIL';
export const LOGOUT_USER = 'LOGOUT_USER';
export const LIKE_ARTICLE_SUCCESS = 'LIKE_ARTICLE_SUCCESS';
export const DISLIKE_ARTICLE_SUCCESS = 'DISLIKE_ARTICLE_SUCCESS';
