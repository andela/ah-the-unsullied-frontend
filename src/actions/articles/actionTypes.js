export const GET_ARTICLE = 'GET_ARTICLE';
export const DELETE_ARTICLE ='DELETE_ARTICLE';