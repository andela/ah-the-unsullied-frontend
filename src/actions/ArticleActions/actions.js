import { toast } from 'react-toastify';
import { GET_ARTICLES } from './types';
import { GET_ERRORS } from '../actionTypes';
import axiosConfig from '../../axiosConfig';

export const getArticles = (page = 1) => {
  return dispatch => {
    return axiosConfig
      .request({
        method: 'get',
        url: `/articles?page=${page}`
      })
      .then(response => {
        dispatch({
          type: GET_ARTICLES,
          payload: response.data.articles
        });
      })
      .catch(err => {
        toast.error(err.response.data.errors.error[0]);
        dispatch({
          type: GET_ERRORS,
          payload: err.response.data
        });
      });
  };
};
