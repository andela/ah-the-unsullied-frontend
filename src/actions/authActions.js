import { toast } from 'react-toastify';
import { SIGNUP_USER, SIGNUP_ERROR } from './actionTypes';
import axiosConfig from '../axiosConfig';

export const registerUser = newUserData => dispatch => {
  const userData = {
    user: newUserData
  };

  return axiosConfig
    .request({
      method: 'post',
      url: 'users/',
      data: userData
    })
    .then(response => {
      dispatch({
        type: SIGNUP_USER,
        payload: response.data
      });
      toast.success(
        'You have successfully created an account,please check your email to verify your account.'
      );
    })
    .catch(err => {
      dispatch({
        type: SIGNUP_ERROR,
        payload: err.response.data
      });
    });
};
