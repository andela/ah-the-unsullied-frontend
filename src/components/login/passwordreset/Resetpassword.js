import React, { Component } from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';
import {resetPassword} from '../../../actions/resetPasswordAction';
import { logoutUser } from '../../../actions/loginActions';

class Resetpassword extends Component {
  state ={
    password:'',
    confirm_password:'',
  }
  handleChange = (e)=>{
    this.setState({
      [e.target.id]: e.target.value
    })
  }
  handleSubmit = (e)=>{
    e.preventDefault();
    const new_password= this.state.password;
    const confirm_password= this.state.confirm_password;
    const token= this.props.match.params.token ;
    this.props.resetPassword(new_password,confirm_password, token);
    this.inputPassword.value=null;
    this.inputPassconfirm.value=null;
  }
  render() {
        const passwordErrorMessage = this.props.errors.errors
        ? this.props.errors.errors.password
        : '';
        const confirmpasswordErrorMessage = this.props.errors.errors
        ? this.props.errors.errors.confirm_password
        : '';
        const passwordmismatchMessage = this.props.errors.message
        ? this.props.errors.message
        : '';
        const successmessage = this.props.message
        ? this.props.message['message']
        : '';
        const success = this.props.success
        ? this.props.success
        : '';
        if (success){
            setTimeout(()=>{
            this.props.history.push('/login')
            },3000);
            this.props.logoutUser();
        }
    return (
      <div>
        <div className='homediv'>
          <h2>Enter New Password</h2>
          <form className='emailform' onSubmit={this.handleSubmit}>

            <label htmlFor='password'>Enter new password:</label>

            <input className='input' type='password' id='password' ref={el => this.inputPassword = el}
                   pattern='(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%&]).{8,}'
                   title='Must contain at least one number and one uppercase and
                    lowercase letter and a special character, and at least 8 or more characters'
                   onChange={this.handleChange} placeholder='Enter your new password..' require/>

            <label htmlFor='msgerror' style={{ color: 'red' }}>{passwordErrorMessage}</label>
            <br/><br/>
            <label htmlFor='password'>Confirm  password:</label>
            <input type='password' ref={el => this.inputPassconfirm = el}
                   onChange={this.handleChange} id='confirm_password'
                   placeholder='confirm password..' require/>
            <label htmlFor='error' style={{ color: 'red' }}>{confirmpasswordErrorMessage}</label>
            <label htmlFor='mismatch' style={{ color: 'red' }}>{passwordmismatchMessage}</label>
            <div id='loader-div' style = {{display:'none', 'margin':'auto'}} >
              <div className='progress'>
                <div className='indeterminate' />
              </div>
            </div>
            <br/><br/>

            <label htmlFor='pass' style={{ color: 'green' }}>{successmessage}</label>
            <br/>
            <input type='submit' value='Submit'/>
          </form>
        </div>
      </div>
    )
  }
}
const mapStateToProps=(state)=>{
  return{
    ...state,
    message:state.passwordreset.message,
    errors:state.passwordreset.errors,
    success:state.passwordreset.success
  }

}
Resetpassword.propTypes = {
  message: PropTypes.func.isRequired,
  errors: PropTypes.func.isRequired,
  success: PropTypes.func.isRequired,
  history: PropTypes.func.isRequired,
  resetPassword: PropTypes.func.isRequired,

    };
export default connect(mapStateToProps, {resetPassword, logoutUser})(Resetpassword)
