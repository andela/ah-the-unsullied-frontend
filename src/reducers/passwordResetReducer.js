import { PASSWORD_RESET, PASSWORD_RESET_ERROR,SEND_EMAIL_ERROR,SEND_EMAIL } from '../actions/actionTypes';

const initialState = {
    message: {},
    errors: {},
    success:false,
    sendemail:'',
    sendemailerr:{},
  };
  
  const passwordResetReducer = (state = initialState, action) => {
    switch (action.type) {
        
      case PASSWORD_RESET:
        return {
          ...state,
          message: action.payload,
          success:true
        };
      case PASSWORD_RESET_ERROR:
        return {
          errors: action.payload
        };
        case SEND_EMAIL:
        return {
          ...state,
          sendemail: action.payload,
          success:true
        };
        case SEND_EMAIL_ERROR:
        return {
          ...state,
          sendemailerr: action.payload,
        };
      default:
        return state;
    }
  };
  
  export default passwordResetReducer;