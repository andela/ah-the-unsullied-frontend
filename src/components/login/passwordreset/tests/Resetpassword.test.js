import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import expect from 'expect';
import Adapter from 'enzyme-adapter-react-16';
import {mapStateToProps, Resetpassword} from '../Resetpassword';
import * as types from '../../../../actions/actionTypes'
import passwordResetReducer from '../../../../reducers/passwordResetReducer'

Enzyme.configure({ adapter: new Adapter() });
describe('component', () => {
  let props;
  let wrapper;
  let wrapperInstance;
  let state;
  beforeEach(() => {
    
    props={
      message: {},
      errors: {},
      success: false,
      message:'Succesful',
      match:{
        params:{
          token :'hgfukjk',
        },
      },
      
      resetPassword: jest.fn(() => {
        Promise.resolve();
      }),
    }

    wrapper = shallow(<Resetpassword {...props} />);
    wrapperInstance = wrapper.instance();
  });
  it('should render password reset form', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should change the state when calling handleChange', () => {
    const event = {
      target: {
        id: 'password',
        value: '$$#EWDsad'
      }
    };
    wrapperInstance.handleChange(event);
    expect(wrapperInstance.state.password).toEqual(event.target.value);
  });

  it('should initiate the updatePassword action on calling handleSubmit', () => {
    const event = {
      preventDefault: jest.fn()
    };
    wrapperInstance.setState(state);
    wrapperInstance.handleSubmit(event);
    expect(props.resetPassword).toHaveBeenCalled();
  });
  it('renders form succesfully', () => {
    const formContainer = shallow(<Resetpassword className="emailform"/>);
    expect(formContainer.length).toEqual(1);
  });
  it('renders home div succesfully', () => {
    const formContainer2 = shallow(<Resetpassword className="homediv"/>);
    expect(formContainer2.length).toEqual(1);
  });
  it('renders input succesfully', () => {
    const formContainer2 = shallow(<Resetpassword className="input"/>);
    expect(formContainer2.length).toEqual(1);
  });

});
const initialState = {};
const password = {
  password: '12345678A!23'
};
const wrongpassword = {
  password: '1234'
};
const email = {
  email: 'unsullieddevs@gmail.com'
};
const wrongemail = {
  email: 'unsullieddevsgmailcom'
};

describe('password reducer', () => {
  it('should change state of success to true when passed with PASSWORD_RESET', () => {
    const action = {
      type: types.PASSWORD_RESET,
      payload: password
    };
    const newState = passwordResetReducer(initialState, action);
    expect(newState.success).toEqual(true);
  });
  it('should change state of success to true when passed with SEND_EMAIL', () => {
    const action = {
      type: types.SEND_EMAIL,
      payload: email
    };
    const newState = passwordResetReducer(initialState, action);
    expect(newState.success).toEqual(true);
  });
  it('should change state of success to false when passed with PASSWORD_RESET_ERROR', () => {
    const action = {
      type: types.PASSWORD_RESET_ERROR,
      payload: wrongpassword
    };
    const newState = passwordResetReducer(initialState, action);
    expect(newState.success).toEqual(false);
  });

  it('should change state of success to false when passed with SEND_EMAIL_ERROR', () => {
    const action = {
      type: types.SEND_EMAIL_ERROR,
      payload: wrongemail
    };
    const newState = passwordResetReducer(initialState, action);
    expect(newState.success).toEqual(false);
  });

  describe('The mapStateToProps', () => {
    const state = {
      passwordreset: {
        success: false,
        message: '',       
        errors: {
          response: 'This is a response'
        }
      }
    };
    const props = mapStateToProps(state);
    expect(props.success).toEqual(state.passwordreset.success);
    expect(props.response).toEqual(state.passwordreset.response);
  });

});
