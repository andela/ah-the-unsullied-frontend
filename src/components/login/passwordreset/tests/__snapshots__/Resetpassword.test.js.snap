// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component should render password reset form 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Resetpassword
    errors={Object {}}
    match={
      Object {
        "params": Object {
          "token": "hgfukjk",
        },
      }
    }
    message="Succesful"
    resetPassword={[MockFunction]}
    success={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="homediv"
      >
        <h2>
          Enter New Password
        </h2>
        <form
          className="emailform"
          onSubmit={[Function]}
        >
          <label
            htmlFor="password"
          >
            Enter new password:
          </label>
          <input
            className="input"
            id="password"
            onChange={[Function]}
            pattern="(?=.*\\\\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%&]).{8,}"
            placeholder="Enter your new password.."
            require={true}
            title="Must contain at least one number and one uppercase and lowercase letter and a special character, and at least 8 or more characters"
            type="password"
          />
          <label
            htmlFor="msgerror"
            style={
              Object {
                "color": "red",
              }
            }
          >
            
          </label>
          <br />
          <br />
          <label
            htmlFor="password"
          >
            Confirm password:
          </label>
          <input
            id="confirm_password"
            onChange={[Function]}
            placeholder="confirm password.."
            require={true}
            type="password"
          />
          <label
            htmlFor="error"
            style={
              Object {
                "color": "red",
              }
            }
          >
            
          </label>
          <label
            htmlFor="mismatch"
            style={
              Object {
                "color": "red",
              }
            }
          >
            
          </label>
          <div
            id="loader-div"
            style={
              Object {
                "display": "none",
                "margin": "auto",
              }
            }
          >
            <div
              className="progress"
            >
              <div
                className="indeterminate"
              />
            </div>
          </div>
          <br />
          <br />
          <label
            htmlFor="pass"
            style={
              Object {
                "color": "green",
              }
            }
          >
            
          </label>
          <br />
          <input
            type="submit"
            value="Submit"
          />
        </form>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Enter New Password
          </h2>,
          <form
            className="emailform"
            onSubmit={[Function]}
          >
            <label
              htmlFor="password"
            >
              Enter new password:
            </label>
            <input
              className="input"
              id="password"
              onChange={[Function]}
              pattern="(?=.*\\\\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%&]).{8,}"
              placeholder="Enter your new password.."
              require={true}
              title="Must contain at least one number and one uppercase and lowercase letter and a special character, and at least 8 or more characters"
              type="password"
            />
            <label
              htmlFor="msgerror"
              style={
                Object {
                  "color": "red",
                }
              }
            >
              
            </label>
            <br />
            <br />
            <label
              htmlFor="password"
            >
              Confirm password:
            </label>
            <input
              id="confirm_password"
              onChange={[Function]}
              placeholder="confirm password.."
              require={true}
              type="password"
            />
            <label
              htmlFor="error"
              style={
                Object {
                  "color": "red",
                }
              }
            >
              
            </label>
            <label
              htmlFor="mismatch"
              style={
                Object {
                  "color": "red",
                }
              }
            >
              
            </label>
            <div
              id="loader-div"
              style={
                Object {
                  "display": "none",
                  "margin": "auto",
                }
              }
            >
              <div
                className="progress"
              >
                <div
                  className="indeterminate"
                />
              </div>
            </div>
            <br />
            <br />
            <label
              htmlFor="pass"
              style={
                Object {
                  "color": "green",
                }
              }
            >
              
            </label>
            <br />
            <input
              type="submit"
              value="Submit"
            />
          </form>,
        ],
        "className": "homediv",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Enter New Password",
          },
          "ref": null,
          "rendered": "Enter New Password",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                htmlFor="password"
              >
                Enter new password:
              </label>,
              <input
                className="input"
                id="password"
                onChange={[Function]}
                pattern="(?=.*\\\\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%&]).{8,}"
                placeholder="Enter your new password.."
                require={true}
                title="Must contain at least one number and one uppercase and lowercase letter and a special character, and at least 8 or more characters"
                type="password"
              />,
              <label
                htmlFor="msgerror"
                style={
                  Object {
                    "color": "red",
                  }
                }
              >
                
              </label>,
              <br />,
              <br />,
              <label
                htmlFor="password"
              >
                Confirm password:
              </label>,
              <input
                id="confirm_password"
                onChange={[Function]}
                placeholder="confirm password.."
                require={true}
                type="password"
              />,
              <label
                htmlFor="error"
                style={
                  Object {
                    "color": "red",
                  }
                }
              >
                
              </label>,
              <label
                htmlFor="mismatch"
                style={
                  Object {
                    "color": "red",
                  }
                }
              >
                
              </label>,
              <div
                id="loader-div"
                style={
                  Object {
                    "display": "none",
                    "margin": "auto",
                  }
                }
              >
                <div
                  className="progress"
                >
                  <div
                    className="indeterminate"
                  />
                </div>
              </div>,
              <br />,
              <br />,
              <label
                htmlFor="pass"
                style={
                  Object {
                    "color": "green",
                  }
                }
              >
                
              </label>,
              <br />,
              <input
                type="submit"
                value="Submit"
              />,
            ],
            "className": "emailform",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Enter new password:",
                "htmlFor": "password",
              },
              "ref": null,
              "rendered": "Enter new password:",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input",
                "id": "password",
                "onChange": [Function],
                "pattern": "(?=.*\\\\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%&]).{8,}",
                "placeholder": "Enter your new password..",
                "require": true,
                "title": "Must contain at least one number and one uppercase and lowercase letter and a special character, and at least 8 or more characters",
                "type": "password",
              },
              "ref": [Function],
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "htmlFor": "msgerror",
                "style": Object {
                  "color": "red",
                },
              },
              "ref": null,
              "rendered": "",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Confirm password:",
                "htmlFor": "password",
              },
              "ref": null,
              "rendered": "Confirm password:",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "confirm_password",
                "onChange": [Function],
                "placeholder": "confirm password..",
                "require": true,
                "type": "password",
              },
              "ref": [Function],
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "htmlFor": "error",
                "style": Object {
                  "color": "red",
                },
              },
              "ref": null,
              "rendered": "",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "htmlFor": "mismatch",
                "style": Object {
                  "color": "red",
                },
              },
              "ref": null,
              "rendered": "",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="progress"
                >
                  <div
                    className="indeterminate"
                  />
                </div>,
                "id": "loader-div",
                "style": Object {
                  "display": "none",
                  "margin": "auto",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="indeterminate"
                  />,
                  "className": "progress",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "indeterminate",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "htmlFor": "pass",
                "style": Object {
                  "color": "green",
                },
              },
              "ref": null,
              "rendered": "",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "type": "submit",
                "value": "Submit",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="homediv"
        >
          <h2>
            Enter New Password
          </h2>
          <form
            className="emailform"
            onSubmit={[Function]}
          >
            <label
              htmlFor="password"
            >
              Enter new password:
            </label>
            <input
              className="input"
              id="password"
              onChange={[Function]}
              pattern="(?=.*\\\\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%&]).{8,}"
              placeholder="Enter your new password.."
              require={true}
              title="Must contain at least one number and one uppercase and lowercase letter and a special character, and at least 8 or more characters"
              type="password"
            />
            <label
              htmlFor="msgerror"
              style={
                Object {
                  "color": "red",
                }
              }
            >
              
            </label>
            <br />
            <br />
            <label
              htmlFor="password"
            >
              Confirm password:
            </label>
            <input
              id="confirm_password"
              onChange={[Function]}
              placeholder="confirm password.."
              require={true}
              type="password"
            />
            <label
              htmlFor="error"
              style={
                Object {
                  "color": "red",
                }
              }
            >
              
            </label>
            <label
              htmlFor="mismatch"
              style={
                Object {
                  "color": "red",
                }
              }
            >
              
            </label>
            <div
              id="loader-div"
              style={
                Object {
                  "display": "none",
                  "margin": "auto",
                }
              }
            >
              <div
                className="progress"
              >
                <div
                  className="indeterminate"
                />
              </div>
            </div>
            <br />
            <br />
            <label
              htmlFor="pass"
              style={
                Object {
                  "color": "green",
                }
              }
            >
              
            </label>
            <br />
            <input
              type="submit"
              value="Submit"
            />
          </form>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Enter New Password
            </h2>,
            <form
              className="emailform"
              onSubmit={[Function]}
            >
              <label
                htmlFor="password"
              >
                Enter new password:
              </label>
              <input
                className="input"
                id="password"
                onChange={[Function]}
                pattern="(?=.*\\\\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%&]).{8,}"
                placeholder="Enter your new password.."
                require={true}
                title="Must contain at least one number and one uppercase and lowercase letter and a special character, and at least 8 or more characters"
                type="password"
              />
              <label
                htmlFor="msgerror"
                style={
                  Object {
                    "color": "red",
                  }
                }
              >
                
              </label>
              <br />
              <br />
              <label
                htmlFor="password"
              >
                Confirm password:
              </label>
              <input
                id="confirm_password"
                onChange={[Function]}
                placeholder="confirm password.."
                require={true}
                type="password"
              />
              <label
                htmlFor="error"
                style={
                  Object {
                    "color": "red",
                  }
                }
              >
                
              </label>
              <label
                htmlFor="mismatch"
                style={
                  Object {
                    "color": "red",
                  }
                }
              >
                
              </label>
              <div
                id="loader-div"
                style={
                  Object {
                    "display": "none",
                    "margin": "auto",
                  }
                }
              >
                <div
                  className="progress"
                >
                  <div
                    className="indeterminate"
                  />
                </div>
              </div>
              <br />
              <br />
              <label
                htmlFor="pass"
                style={
                  Object {
                    "color": "green",
                  }
                }
              >
                
              </label>
              <br />
              <input
                type="submit"
                value="Submit"
              />
            </form>,
          ],
          "className": "homediv",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Enter New Password",
            },
            "ref": null,
            "rendered": "Enter New Password",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label
                  htmlFor="password"
                >
                  Enter new password:
                </label>,
                <input
                  className="input"
                  id="password"
                  onChange={[Function]}
                  pattern="(?=.*\\\\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%&]).{8,}"
                  placeholder="Enter your new password.."
                  require={true}
                  title="Must contain at least one number and one uppercase and lowercase letter and a special character, and at least 8 or more characters"
                  type="password"
                />,
                <label
                  htmlFor="msgerror"
                  style={
                    Object {
                      "color": "red",
                    }
                  }
                >
                  
                </label>,
                <br />,
                <br />,
                <label
                  htmlFor="password"
                >
                  Confirm password:
                </label>,
                <input
                  id="confirm_password"
                  onChange={[Function]}
                  placeholder="confirm password.."
                  require={true}
                  type="password"
                />,
                <label
                  htmlFor="error"
                  style={
                    Object {
                      "color": "red",
                    }
                  }
                >
                  
                </label>,
                <label
                  htmlFor="mismatch"
                  style={
                    Object {
                      "color": "red",
                    }
                  }
                >
                  
                </label>,
                <div
                  id="loader-div"
                  style={
                    Object {
                      "display": "none",
                      "margin": "auto",
                    }
                  }
                >
                  <div
                    className="progress"
                  >
                    <div
                      className="indeterminate"
                    />
                  </div>
                </div>,
                <br />,
                <br />,
                <label
                  htmlFor="pass"
                  style={
                    Object {
                      "color": "green",
                    }
                  }
                >
                  
                </label>,
                <br />,
                <input
                  type="submit"
                  value="Submit"
                />,
              ],
              "className": "emailform",
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Enter new password:",
                  "htmlFor": "password",
                },
                "ref": null,
                "rendered": "Enter new password:",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "input",
                  "id": "password",
                  "onChange": [Function],
                  "pattern": "(?=.*\\\\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%&]).{8,}",
                  "placeholder": "Enter your new password..",
                  "require": true,
                  "title": "Must contain at least one number and one uppercase and lowercase letter and a special character, and at least 8 or more characters",
                  "type": "password",
                },
                "ref": [Function],
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                  "htmlFor": "msgerror",
                  "style": Object {
                    "color": "red",
                  },
                },
                "ref": null,
                "rendered": "",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Confirm password:",
                  "htmlFor": "password",
                },
                "ref": null,
                "rendered": "Confirm password:",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "confirm_password",
                  "onChange": [Function],
                  "placeholder": "confirm password..",
                  "require": true,
                  "type": "password",
                },
                "ref": [Function],
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                  "htmlFor": "error",
                  "style": Object {
                    "color": "red",
                  },
                },
                "ref": null,
                "rendered": "",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                  "htmlFor": "mismatch",
                  "style": Object {
                    "color": "red",
                  },
                },
                "ref": null,
                "rendered": "",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="progress"
                  >
                    <div
                      className="indeterminate"
                    />
                  </div>,
                  "id": "loader-div",
                  "style": Object {
                    "display": "none",
                    "margin": "auto",
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <div
                      className="indeterminate"
                    />,
                    "className": "progress",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "indeterminate",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                  "type": "div",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                  "htmlFor": "pass",
                  "style": Object {
                    "color": "green",
                  },
                },
                "ref": null,
                "rendered": "",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "type": "submit",
                  "value": "Submit",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "form",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
